### DISCUSSION TITLE
keep basic features from C: #include and static declaration

### DISCUSSION THREAD

**Discussion by massimosala** [koka-lang/koka#298] [General]
Hi
I really like Koka, it has *many* features not available in other languages.

I believe these two features of the C language are essential:
`#include`
`static var = value`
and I suffer when they are not implemented in other languages.

I kindly suggest adding them to koka.

Forgive me if it is already possible to get their functionality, I have read the examples but I cannot find how to do.
Sincerely, Sala
massimo.sala.71@gmail.com

**Comment by timjs** [comment]
Hi Sala,

Koka has a module system and you can use `import` to include functions and values from other source files. 

And with `static` variables you mean constants? Because all `val`s in Koka are already constant.

**Reply by massimosala** [reply]
Hi!

I mean the classic C static declaration inside a function.

This is the simplest example:
```
int counter(void)
{
  static int n = 0;
  n += 1;
  return n;
}

```

At each function call, the return value is incremented by 1.
*static* is used to keep variables in memory between function calls; the variable scope is local to the function.
It is very useful and sometimes "forgotten" in modern languages, sic!

**Reply by CaramelFur** [reply]
Hi, that is indeed a neat feature, I did not know you could do that in C.
But this will never make it into Koka though, as this feature causes undefined side effects. 
There might be a way to do something similar using Koka's side effect handlers, but I'm not sure.
